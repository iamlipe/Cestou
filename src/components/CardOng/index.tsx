import React, {memo} from 'react';
import styled from 'styled-components/native';
import {Linking} from 'react-native';

interface Props {
  title: string;
  image: string;
  link?: string;
  onPress?: () => void;
}

const CardOng = ({title, image, link, onPress}: Props) => {
  return (
    <StyledContainer
      onPress={async () => {
        if (link) Linking.openURL(link);
        if (onPress) onPress();
      }}>
      <StyledImage testID="card-ong-image" source={image} />
      <StyledTitle>{title}</StyledTitle>
    </StyledContainer>
  );
};

const StyledContainer = styled.TouchableOpacity`
  width: 48%;
  align-items: center;
`;

const StyledImage = styled.Image`
  width: 100%;
  height: 130px;
  border-radius: 4px;

  margin-bottom: 5px;
`;

const StyledTitle = styled.Text`
  font-family: ${({theme}) => theme.fonts.SEMIBOLD_SOURCESANSPRO};
  font-size: ${({theme}) => theme.sizing.SMALLER};
  color: ${({theme}) => theme.colors.GRAY_900};
`;

export default memo(CardOng);
