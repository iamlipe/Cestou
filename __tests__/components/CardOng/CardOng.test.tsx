import React from 'react';
import {
  renderWithThemeProvider,
  matchSnapshotWithProvider,
} from '@/config/functions/Testing';

import imgOng from '@/assets/images/ong_1.png';
import CardOng from '@/components/CardOng';

describe('CardOng', () => {
  const title = 'ong';
  const image = imgOng;

  test('should render correctly', () => {
    renderWithThemeProvider(<CardOng title={title} image={image} />);
  });

  test('should matches snapshot', () => {
    matchSnapshotWithProvider(<CardOng title={title} image={image} />);
  });

  test('should render title', () => {
    const cardOng = renderWithThemeProvider(
      <CardOng title={title} image={image} />,
    );

    const cardTitle = cardOng.queryByText(title);

    expect(cardTitle).toBeTruthy();
  });

  test('should render image', () => {
    const cardOng = renderWithThemeProvider(
      <CardOng title={title} image={image} />,
    );

    const cardImage = cardOng.queryByTestId('card-ong-image');

    expect(cardImage).toBeTruthy();
  });
});
