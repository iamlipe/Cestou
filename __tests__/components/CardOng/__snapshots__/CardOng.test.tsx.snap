// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardOng should matches snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "opacity": 1,
      "width": "48%",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/ong_1.png",
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 4,
          "height": 130,
          "marginBottom": 5,
          "width": "100%",
        },
      ]
    }
    testID="card-ong-image"
  />
  <Text
    style={
      Array [
        Object {
          "color": "#1A1A1A",
          "fontFamily": "SourceSansPro-SemiBold",
          "fontSize": 27,
        },
      ]
    }
  >
    ong
  </Text>
</View>
`;
